src/Host.cpp

```diff
@@ -921,22 +921,25 @@ std::tuple<bool, QString, QString> Host::saveProfile(const QString& saveFolder,
 
     if (saveFolder.isEmpty() && saveName.isEmpty()) {
         // This is likely to be the save as the profile is closed
-        qDebug().noquote().nospace() << "Host::saveProfile(...) INFO - called with no saveFolder or saveName arguments for profile '"
-                                     << mHostName
+        qDebug().noquote().nospace() << "Host::saveProfile(...) INFO - called with no saveFolder or saveName arguments for profile '" << mHostName
                                      << "' so assuming it is an end of session save and the TCommandLines' histories need saving...";
         emit signal_saveCommandLinesHistory();
     }
 
-    emit profileSaveStarted();
-    qApp->processEvents();
-
     auto writer = new XMLexport(this);
     writers.insert(qsl("profile"), writer);
     writer->exportHost(filename_xml);
     mWritingHostAndModules = true;
+
+    // emit signal to notify the UI that the save button should get disabled momentarily
+    // this needs to run after `writers` and `mWritingHostAndModules` have been set
+    // so that the currentlySavingProfile() check can run properly
+    emit profileSaveStarted();
+    qApp->processEvents();
```

Comment:
I don't this change is doing what you say/think it does - the `emit` only causes the editor save buttons to be disabled. The prior `mWritingHostAndModules = true;` is what causes subsequent calls to `Host::saveProfile(...)` to bail out - and TBH I ***suspect*** it should be set BEFORE the `XMLexport::exportHost(const QString&)` is called!

*I am guessing but I think that the relocation of the `qApp-->processEvents();` has modified the way things are working to the extent that it seems to have cured the crashes...*

Reply:
Yes, this is what I really meant.

Reply:
Do you think the `mWritingHostAndModules = true;` should be repositioned before the `writer->exportHost(filename_xml);` and the two lines you moved could be put back where they were (or put before the `writer->exportHost(filename_xml);`) as well?

Reply:
Might work either way, feel free to test it out